### PR TITLE
fix: add missing general pagination test case

### PR DESCRIPTION
**Pull request by leongold**
- pagination test case for non-azure repositories was missing and was added
- this test cased exposed what seems to be an ECR bug, which is addressed in the PR
- prettier formatting
- improved descriptions

### CODE REVIEW

File: test/src/get-tags.test.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ describe("tags pagination", () => {
 
 for (const registry of registries) {
   describe(registry.name, () => {
-    test("valid credentials and repository returns tags", async () => {
+    test("valid non-paginated request", async () => {
```
Comment (leongold) [inline test/src/get-tags.test.ts]: This test was missing.

File: test/src/get-tags.test.ts [hunk 2]
```diff
@@ -56,6 +56,29 @@ for (const registry of registries) {
       expect(typeof exampleTag).toBe("string");
     });
 
+    if (registry.name !== "ElasticCR") {
```
Comment (leongold) [inline test/src/get-tags.test.ts]: I looked & failed to find a better way to programatically skip a test in jest.

File: test/src/get-tags.test.ts [hunk 3]
```diff
@@ -6,11 +6,11 @@ jest.setTimeout(100000);
 
 const registries = [AzureCR, ElasticCR];
 
-describe("tags pagination", () => {
+describe("tags pagination of a large (>1000, automatically paginated) repository", () => {
```
Comment (leongold) [inline test/src/get-tags.test.ts]: Better description; Azure's Ubunu repo contains roughly 5k~ images which allows us to test behaviour under automatic pagination.

File: test/src/get-tags.test.ts [hunk 4]
```diff
@@ -56,6 +56,29 @@ for (const registry of registries) {
       expect(typeof exampleTag).toBe("string");
     });
 
+    if (registry.name !== "ElasticCR") {
+      // ECR pagination is currently broken, see /src/get-tags.ts
+      test("valid paginated request", async done => {
```
Comment (leongold) [inline test/src/get-tags.test.ts]: this test was missing

File: src/get-tags.ts [hunk 5]
```diff
@@ -11,14 +11,44 @@ export async function getTags(
...
+      endpoint,
+      "tags",
+      pageSize,
+      maxPages
+    );
+  } catch (err) {
+    if (err.statusCode !== 405) {
+      throw err;
+    }
+    /// ECR seemingly do not properly implement pagination. healthy link header example (ACR):
```
Comment (mladkau) [inline src/get-tags.ts]: Oh dear ... I shall reconsider my statement of Amazon being good :-)

**Pull request merged**